File: ender-does-backend-NET/User/Mappers/UserMapper.cs
using ender_does_backend_NET.Todo.Mappers;
using ender_does_backend_NET.User.DTOs;

namespace ender_does_backend_NET.User.Mappers;

public static class UserMapper
{
    public static UserResponse ToResponse(Models.User user)
    {
        return new UserResponse(
            user.Id,
            user.Name,
            user.Email,
            user.ImageUrl,
            user.AccountLocked,
            user.Enabled,
            [
                .. user.Todos
                    .Select(TodoMapper.ToResponse)
            ]
        );
    }
}

File: ender-does-backend/src/main/java/com/enderdev/enderdoesbackend/user/mappers/UserMapper.java
package com.enderdev.enderdoesbackend.user.mappers;

import com.enderdev.enderdoesbackend.todo.mappers.TodoMapper;
import com.enderdev.enderdoesbackend.user.dto.UserResponse;
import com.enderdev.enderdoesbackend.user.models.Role;
import com.enderdev.enderdoesbackend.user.models.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class UserMapper {
    private final TodoMapper todoMapper;
    public UserResponse toUserResponse(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .imageUrl(user.getImageUrl())
                .accountLocked(user.getAccountLocked())
                .enabled(user.getEnabled())
                .roles(user.getRoles().stream().map(Role::getName).toList())
                .todos(user.getTodos().stream().map(todoMapper::toTodoResponse).toList())
                .build();
    }
}
